### PR TITLE
Implement 'hello' interactive command-line tool.

### PR DESCRIPTION
**Pull request by jtchitty**
Implement terminal configuration and I/O strategy.
Create basic menu functionality.
Implement LED control screen.
Implement Buzzer control screen (w/o period and percent control).

### CODE REVIEW

**Comment by jtchitty** [comment]
Sadly, I couldn't use ncurses, because of some issues with the way that commands provided by legato apps are launched.  In particular, the terminal is not passed through to them in such a way that ncurses can properly manage the terminal.  We'll have to leave that for the future.

File: apps/Welcome/components/yellowCommandLine/mainMenu.c [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * The Screen object for the main menu.
+ */
+//--------------------------------------------------------------------------------------------------
+static Screen_t MainMenu =
+{
+    drawFunc: DrawMainMenu,
```
Comment (dpfrey) [inline apps/Welcome/components/yellowCommandLine/mainMenu.c]: This type of initialization is GCC-only and isn't suggested for use given that there's a standard way to achieve the same thing. https://gcc.gnu.org/onlinedocs/gcc-9.1.0/gcc/Designated-Inits.html#Designated-Inits
`drawFunc: DrawMainMenu,` should be changed to `.drawFunc = DrawMainMenu,`

File: apps/Welcome/components/yellowCommandLine/buzzerScreen.c [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            printf("Sorry, period configuration is not yet implemented.\n");
+            break;
+
+        case '3':
+            printf("Sorry, duty cycle percentage configuration is not yet implemented.\n");
+            break;
+
+        default:
+
+            // Eat the invalid selection.
```
[the remark this answers is not in the document]
Comment (jtchitty) [reply apps/Welcome/components/yellowCommandLine/buzzerScreen.c]: OK.  Will do in next rev.

File: apps/Welcome/components/yellowCommandLine/mainMenu.c (hunk 1 above)
Comment (jtchitty) [reply apps/Welcome/components/yellowCommandLine/mainMenu.c]: Will do in next rev.

File: apps/Welcome/components/yellowCommandLine/cmdLine.c [hunk 3]
```diff
@@ -60,8 +79,37 @@ static enum
...
+typedef struct
+{
+    char text[80];   ///< The menu entry's text string (null-terminated).
+    cmdLine_MenuSelectionFunc selectionFunc; ///< Function to be called when entry is selected.
+    void* contextPtr;   ///< Opaque pointer value to be passed to the selectionFunc.
+}
+MenuEntry_t;
+
+/// List of Menu Entries.  Index is menu option number minus 1.
+static MenuEntry_t MenuEntryList[9];
```
Comment (dpfrey) [inline apps/Welcome/components/yellowCommandLine/cmdLine.c]: Can you create a `MAX_MENU_OPTIONS` define?  The literal `9` appears in a few different places in the code.

**Pull request merged**